src/commands/periscope/helpers/storageandhtmlhelper.ts

```diff
@@ -0,0 +1,170 @@
+import * as vscode from 'vscode';
+import { getSASKey } from '../../utils/host';
+import { parseResource } from '../../../azure-api-utils';
+import * as ast from 'azure-arm-storage';
+import { PeriscopeStorage } from '../models/storage';
+import * as amon from 'azure-arm-monitor';
+import * as htmlhandlers from "handlebars";
+import * as path from 'path';
+import * as fs from 'fs';
+
+const {
+    BlobServiceClient,
+    StorageSharedKeyCredential
+} = require("@azure/storage-blob");
+
+export async function getClusterDiagnosticSettings(target: any) {
+    try {
+        // Get daignostic setting via diagnostic monitor
+        const diagnosticMonitor = new amon.MonitorManagementClient(target.root.credentials, target.root.subscriptionId);
+        const diagnosticSettings = await diagnosticMonitor.diagnosticSettingsOperations.list(target.id);
+
+        return diagnosticSettings;
+    } catch (e) {
+        vscode.window.showErrorMessage(`Can't: ${e}`);
+        return undefined;
+    }
+}
+
+export async function getStorageInfo(
+    target: any,
+    diagnosticStorageAccountId: string
+): Promise<PeriscopeStorage | undefined> {
+    try {
+        const { resourceGroupName, name: accountName } = parseResource(diagnosticStorageAccountId!);
+
+        if (!resourceGroupName || !accountName) {
+            vscode.window.showErrorMessage(`Invalid storage id ${diagnosticStorageAccountId} associated with the cluster`);
+            return;
+        }
+
+        // Get keys from storage client.
+        const storageClient = new ast.StorageManagementClient(target.root.credentials, target.root.subscriptionId);
+        const storageAccKeyList = await storageClient.storageAccounts.listKeys(resourceGroupName, accountName);
+        const accountKey = storageAccKeyList?.keys!.find((it) => it.keyName === "key1")?.value;
+
+        // Generate SAS.
+        const sas = getSASKey(accountName, accountKey!);
+
+        return <PeriscopeStorage>{ storageName: accountName, storageKey: accountKey, storageDeploymentSas: sas };
+    } catch (e) {
+        vscode.window.showErrorMessage(`Storage associated with cluster had following error: ${e}`);
+        return undefined;
+    }
+}
+
+export async function generateDownloadableLinks(
+    periscopeStorage: PeriscopeStorage,
+    output: string,
+    generateSevenDaySas = false) {
```

Comment:
💡 This will be changes to `expiryDate` soon, following one of the idea you mentioned, that way this functionalists is more usable by anyone passing the expiry date and getting the `sas` back.